src/uk/gov/hmcts/contino/GradleBuilder.groovy

```diff
@@ -88,10 +100,56 @@ class GradleBuilder extends AbstractBuilder {
       }
       finally {
         steps.archiveArtifacts 'build/reports/dependency-check-report.html'
+        publishCVEReport()
       }
     }
   }
 
+  def publishCVEReport() {
+    try {
+      steps.withCredentials([[$class: 'StringBinding', credentialsId: 'COSMOSDB_TOKEN_KEY', variable: 'COSMOSDB_TOKEN_KEY']]) {
+        if (steps.env.COSMOSDB_TOKEN_KEY == null) {
+          steps.echo "Set the 'COSMOSDB_TOKEN_KEY' environment variable to enable metrics publishing"
+          return
+        }
+
+        steps.echo "Publishing CVE report"
+        String dependencyReport = steps.readFile('build/reports/dependency-check-report.json')
+        def summary = prepareCVEReport(dependencyReport, steps.env)
+        createDocument(summary)
+      }
+    } catch (err) {
+      steps.echo "Unable to publish CVE report '${err}'"
+    }
+  }
+
+  def prepareCVEReport(owaspReportJSON, env) {
+    def report = new JsonSlurper().parseText(owaspReportJSON)
+    // Only include non-vulnerable dependencies to reduce the report size; Cosmos has a 2MB limit.
+    report.dependencies = report.dependencies.findAll { it.vulnerabilityIds }
+
+    def result = [
+      build: [
+        build_display_name           : env.BUILD_DISPLAY_NAME,
+        build_tag                    : env.BUILD_TAG,
+        git_url                      : env.GIT_URL,
+      ],
+      report: report
+    ]
+    return JsonOutput.toJson(result)
+  }
+
+  @NonCPS
+  private def createDocument(String reportJSON) {
+    def client = new DocumentClient(cosmosDbUrl, steps.env.COSMOSDB_TOKEN_KEY, null, null)
```

Comment:
isn't this autocloseable?
https://docs.microsoft.com/en-us/java/api/com.microsoft.azure.documentdb.documentclient?view=azure-java-stable (doesn't map to the version we're using so it might not)

Reply:
withCloseable for AutoCloseable is since Groovy 2.5.0 - http://docs.groovy-lang.org/docs/latest/html/groovy-jdk/java/lang/AutoCloseable.html